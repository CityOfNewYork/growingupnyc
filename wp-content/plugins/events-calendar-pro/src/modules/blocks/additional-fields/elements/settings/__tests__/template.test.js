/**
 * Internal dependencies
 */
import React from 'react';
import Settings from '../template';

describe( '<Settings>', () => {
	test( 'no link is provided', () => {
		const component = renderer.create(
			<Settings name={ 'Additional Fields' } />,
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'link is provided', () => {
		const component = renderer.create(
			<Settings
				name={ 'Additional Fields' }
				settingsLink="http://guti.local/wp-admin/edit.php?page=tribe-common&tab=additional-fields&post_type=tribe_events"
			/>,
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'link is provided with before', () => {
		const component = renderer.create(
			<Settings
				name={ 'Additional Fields' }
				settingsLink="http://guti.local/wp-admin/edit.php?page=tribe-common&tab=additional-fields&post_type=tribe_events"
				before={ <span>Before</span> }
			/>,
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'link is provided with after', () => {
		const component = renderer.create(
			<Settings
				name={ 'Additional Fields' }
				settingsLink="http://guti.local/wp-admin/edit.php?page=tribe-common&tab=additional-fields&post_type=tribe_events"
				after={ <span>after</span> }
			/>,
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
