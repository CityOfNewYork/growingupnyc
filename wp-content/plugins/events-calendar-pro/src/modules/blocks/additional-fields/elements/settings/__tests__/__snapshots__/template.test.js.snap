// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings> link is provided 1`] = `
<div
  className="InspectorControls"
>
  <span
    className="PanelBody"
  >
    <span>
      Adjust this block’s options under Events → Settings → 
      <a
        href="http://guti.local/wp-admin/edit.php?page=tribe-common&tab=additional-fields&post_type=tribe_events"
        rel="noreferrer noopener"
        target="_blank"
      >
        Additional Fields
      </a>
    </span>
  </span>
</div>
`;

exports[`<Settings> link is provided with after 1`] = `
<div
  className="InspectorControls"
>
  <span
    className="PanelBody"
  >
    <span>
      Adjust this block’s options under Events → Settings → 
      <a
        href="http://guti.local/wp-admin/edit.php?page=tribe-common&tab=additional-fields&post_type=tribe_events"
        rel="noreferrer noopener"
        target="_blank"
      >
        Additional Fields
      </a>
    </span>
  </span>
  <span>
    after
  </span>
</div>
`;

exports[`<Settings> link is provided with before 1`] = `
<div
  className="InspectorControls"
>
  <span>
    Before
  </span>
  <span
    className="PanelBody"
  >
    <span>
      Adjust this block’s options under Events → Settings → 
      <a
        href="http://guti.local/wp-admin/edit.php?page=tribe-common&tab=additional-fields&post_type=tribe_events"
        rel="noreferrer noopener"
        target="_blank"
      >
        Additional Fields
      </a>
    </span>
  </span>
</div>
`;

exports[`<Settings> no link is provided 1`] = `
<div
  className="InspectorControls"
>
  <span
    className="PanelBody"
  />
</div>
`;
